packages/back-end/src/services/experimentTimeSeries.ts

```diff
@@ -0,0 +1,98 @@
+import md5 from "md5";
+import { getAllMetricIdsFromExperiment } from "shared/experiments";
+import { ReqContext } from "back-end/types/organization";
+import { ExperimentInterface } from "back-end/src/validators/experiments";
+import {
+  ExperimentSnapshotInterface,
+  SnapshotMetric,
+} from "back-end/types/experiment-snapshot";
+import {
+  MetricTimeSeriesDataPoint,
+  MetricTimeSeriesVariation,
+} from "back-end/src/validators/metric-time-series";
+
+export async function updateExperimentTimeSeries({
+  context,
+  experiment,
+  experimentSnapshot,
+}: {
+  context: ReqContext;
+  experiment: ExperimentInterface;
+  experimentSnapshot: ExperimentSnapshotInterface;
+}) {
+  const metricsIds = getAllMetricIdsFromExperiment(experiment);
+  const relativeAnalysis = experimentSnapshot.analyses.find(
+    (analysis) => analysis.settings.differenceType === "relative"
+  );
+  const absoluteAnalysis = experimentSnapshot.analyses.find(
+    (analysis) => analysis.settings.differenceType === "absolute"
+  );
+  const scaledAnalysis = experimentSnapshot.analyses.find(
+    (analysis) => analysis.settings.differenceType === "scaled"
+  );
+
+  // NB: Using relative as a base, but it should match absolute & scaled
```

Comment:
This is where the `value`, `denominator` and `stats` fields could also come from relative and match absolute and scaled if we want to refactor the data model to make it even leaner.